lib/api/BackbeatAPI.js

```diff
@@ -560,17 +615,21 @@ class BackbeatAPI {
         return async.eachLimit(reqBody, 10, (o, next) => {
             const { Bucket, Key, VersionId, StorageClass } = o;
             const key = getFailedCRRKey(Bucket, Key, VersionId, StorageClass);
-            const cmd = ['get', key];
-            return this._redis.batch([cmd], (err, res) => {
-                if (err) {
-                    return next(err);
-                }
-                const [cmdErr, sourceRole] = res[0];
-                // If the key did not exist, value is `null`.
-                if (sourceRole !== null) {
-                    entries.push(new ObjectFailureEntry(key, sourceRole));
-                }
-                return next(cmdErr);
+            return this._redis.scan(`${key}:*`, undefined, (err, keys) => {
+                // There can only be one failed key. We are only using the glob
+                // because we do not know the original timestamp.
+                const [key] = keys;
+                return this._redis.batch([['get', key]], (err, res) => {
```

Comment:
Why use batch when you are only accessing `get`? Shouldn't `this._redis.get` just work?

Reply:
`this._redis` is an instance of the `RedisClient` wrapper imported from Arsenal and it only exposes certain methods. It would be cleaner to add those wrapper methods though that would work with `ioredis`. Perhaps this could be a follow-up refactor?